src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -153,7 +149,10 @@ boolean isConnectionAlive(final Connection connection)
    {
       try {
          try {
-            setNetworkTimeout(connection, validationTimeout);
+            boolean timeoutSet = trySetNetworkTimeout(connection, validationTimeout);
+            if (!timeoutSet) {
+               return false;
+            }
```

Comment:
It's hard to describe, but something about this code just seems off to me.

The original code seems more correct, in that it would throw an exception if there is an error. This I believe is the real exception that the original issue is about. This is the exception that you want to be seing.

The reason it is eaten is because the setNetworkTimeout call in the finally below would then throw another exception.

My gut tells me that this call should actually be one line up, just before the try rather than inside the try block. This way if the first setNetworkTimeout call throws, it won't try to call setNetworkTimeout a second time.

But then the rollback call at the bottom of the finally wouldn't get called. I don't know if it would need to be, but it is a change in behavior.

So actually it might just be that there should be one more try/finally added to this method so that the rollback call can be in a finally all by itself.